velox/functions/sparksql/tests/JsonFunctionsTest.cpp

```diff
@@ -0,0 +1,119 @@
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/type/Type.h"
+
+#include <stdint.h>
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class JsonFunctionTest : public SparkFunctionBaseTest {
+ protected:
+  std::optional<std::string> getJsonObject(
+      const std::optional<std::string>& json,
+      const std::optional<std::string>& jsonPath) {
+    return evaluateOnce<std::string>("get_json_object(c0, c1)", json, jsonPath);
+  }
+};
+
+TEST_F(JsonFunctionTest, getJsonObject) {
+  EXPECT_EQ(getJsonObject(R"({"hello": "3.5"})", "$.hello"), "3.5");
+  EXPECT_EQ(getJsonObject(R"({"hello": 3.5})", "$.hello"), "3.5");
+  EXPECT_EQ(getJsonObject(R"({"hello": 292222730})", "$.hello"), "292222730");
+  EXPECT_EQ(getJsonObject(R"({"hello": -292222730})", "$.hello"), "-292222730");
+  EXPECT_EQ(getJsonObject(R"({"my": {"hello": 3.5}})", "$.my.hello"), "3.5");
+  EXPECT_EQ(getJsonObject(R"({"my": {"hello": true}})", "$.my.hello"), "true");
+  EXPECT_EQ(getJsonObject(R"({"hello": ""})", "$.hello"), "");
+  EXPECT_EQ(
+      "0.0215434648799772",
+      getJsonObject(R"({"score":0.0215434648799772})", "$.score"));
+  // Returns input json if json path "$".
+  EXPECT_EQ(
+      getJsonObject(R"({"name": "Alice", "age": 5, "id": "001"})", "$"),
+      R"({"name": "Alice", "age": 5, "id": "001"})");
+  EXPECT_EQ(
+      getJsonObject(R"({"name": "Alice", "age": 5, "id": "001"})", "$.age"),
+      "5");
+  EXPECT_EQ(
+      getJsonObject(R"({"name": "Alice", "age": 5, "id": "001"})", "$.id"),
+      "001");
+  EXPECT_EQ(
+      getJsonObject(
+          R"([{"my": {"param": {"name": "Alice", "age": "5", "id": "001"}}}, {"other": "v1"}])",
+          "$[0]['my']['param']['age']"),
+      "5");
+  EXPECT_EQ(
+      getJsonObject(
+          R"([{"my": {"param": {"name": "Alice", "age": "5", "id": "001"}}}, {"other": "v1"}])",
+          "$[0].my.param.age"),
+      "5");
+
+  // Json object as result.
+  EXPECT_EQ(
+      getJsonObject(
+          R"({"my": {"param": {"name": "Alice", "age": "5", "id": "001"}}})",
+          "$.my.param"),
+      R"({"name": "Alice", "age": "5", "id": "001"})");
+  EXPECT_EQ(
+      getJsonObject(
+          R"({"my": {"param": {"name": "Alice", "age": "5", "id": "001"}}})",
+          "$['my']['param']"),
+      R"({"name": "Alice", "age": "5", "id": "001"})");
+
+  // Array as result.
+  EXPECT_EQ(
+      getJsonObject(
+          R"([{"my": {"param": {"name": "Alice"}}}, {"other": ["v1", "v2"]}])",
+          "$[1].other"),
+      R"(["v1", "v2"])");
+  // Array element as result.
+  EXPECT_EQ(
+      getJsonObject(
+          R"([{"my": {"param": {"name": "Alice"}}}, {"other": ["v1", "v2"]}])",
+          "$[1].other[0]"),
+      "v1");
+  EXPECT_EQ(
+      getJsonObject(
+          R"([{"my": {"param": {"name": "Alice"}}}, {"other": ["v1", "v2"]}])",
+          "$[1].other[1]"),
+      "v2");
+
+  // Field not found.
```

Comment:
This test is rather long. Consider extracting invalid cases into a separate test method.